src/definitions/constants.rs

```diff
@@ -17,10 +17,23 @@ pub(crate) const CONSUMED_MSG_TO_L2_ENCODED_DATA_SIZE: usize =
 pub(crate) const LOG_MSG_TO_L1_ENCODED_DATA_SIZE: usize =
     (L2_TO_L1_MSG_HEADER_SIZE + 1) - LOG_MSG_TO_L1_N_TOPICS;
 
+/// The (empirical) L1 gas cost of each Cairo step.
+pub(crate) const N_STEPS_FEE_WEIGHT: f64 = 0.01;
+
 lazy_static! {
-    // TODO: should have an additional (builtin, 0.0) for each builtin (except for keccak)
+    // Ratios are taken from the `starknet_instance` CairoLayout object in cairo-lang.
     pub static ref DEFAULT_CAIRO_RESOURCE_FEE_WEIGHTS: HashMap<String, f64> =
-        HashMap::from([("n_steps".to_string(), 1.0)]);
+        HashMap::from([
+            ("n_steps".to_string(), N_STEPS_FEE_WEIGHT),
+            ("output_builtin".to_string(), 0.0),
+            ("pedersen_builtin".to_string(), N_STEPS_FEE_WEIGHT * 32.0),
+            ("range_check_builtin".to_string(), N_STEPS_FEE_WEIGHT * 16.0),
+            ("ecdsa_builtin".to_string(), N_STEPS_FEE_WEIGHT * 2048.0),
+            ("bitwise_builtin".to_string(), N_STEPS_FEE_WEIGHT * 64.0),
+            ("ec_op_builtin".to_string(), N_STEPS_FEE_WEIGHT * 1024.0),
+            ("poseidon_builtin".to_string(), N_STEPS_FEE_WEIGHT * 32.0),
+            ("segment_arena_builtin".to_string(), N_STEPS_FEE_WEIGHT * 10.0),
```

Comment:
That's a good question -- for some reason "segment_arena" builtin isn't in the [ALL_BUILTINS list](https://github.com/starkware-libs/cairo-lang/blob/c954f154bbab04c3fb27f7598b015a9475fc628e/src/starkware/cairo/lang/builtins/all_builtins.py#L50)...

Reply:
I think it is outside the ALL_BUILTINS list because in cairo-lang it is not part of the vm code but passed as a parameter when running an execution entrypoint like for example the syscall_handler